### PR TITLE
Support page generation

### PR DESCRIPTION
**Pull request by seancpeters**
There are still things to work out regarding the conditions and locations for generating these files. See #763

### CODE REVIEW

File: src/VS.Web.CG.Mvc/VS.Web.CG.Mvc.csproj [hunk 1]
```diff
@@ -42,9 +44,18 @@
     <None Pack="true" Include="Templates\Identity\wwwroot\lib\jquery-validation\dist\*" PackagePath="Templates\Identity\wwwroot\lib\jquery-validation\dist\" />
     <None Pack="true" Include="Templates\Identity\wwwroot\lib\jquery-validation-unobtrusive\*" PackagePath="Templates\Identity\wwwroot\lib\jquery-validation-unobtrusive\" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Remove="Templates\Identity\SupportPages\**" />
```
Comment (prafullbhosale) [inline src/VS.Web.CG.Mvc/VS.Web.CG.Mvc.csproj]: Might be good to exclude the entire Templates folder here.

File: src/VS.Web.CG.Mvc/Identity/identitygeneratorfilesconfig.json [hunk 2]
```diff
@@ -511,6 +503,295 @@
...
+        "ShowInListFiles": false
+      },
+      {
+        "Name": "Banner3.svg",
+        "SourcePath": "wwwroot/images/banner3.svg",
+        "OutputPath": "wwwroot/images/banner3.svg",
+        "IsTemplate": false,
+        "ShowInListFiles": false
+      },
+      {
```
Comment (javiercn) [inline src/VS.Web.CG.Mvc/Identity/identitygeneratorfilesconfig.json]: Site.js and Site.min.js I don't think you need (I believe they are empty, you might need to remove the reference from _Layout.cshtml or similar if there is any)

**Comment by javiercn** [comment]
@seancpeters Please work with Nate or @mkArtakMSFT to figure out where and when this does have to be merged.

File: src/VS.Web.CG.Mvc/Identity/IdentityGeneratorFilesConfig.cs [hunk 3]
```diff
@@ -105,12 +97,42 @@ internal static IdentityGeneratorFile[] GetFilesToGenerate(IEnumerable<string> n
...
+                filesToGenerate.Add(layout);
+            }
+            else
+            {
+                IdentityGeneratorFile validationScriptsPartial = new IdentityGeneratorFile()
+                {
+                    Name = "_ValidationScriptsPartial",
+                    SourcePath = "Pages/_ValidationScriptsPartial.cshtml",
+                    OutputPath = "Areas/Identity/Pages/_ValidationScriptsPartial.cshtml",
+                    IsTemplate = false
```
[the remark this answers is not in the document]
Comment (seancpeters) [reply src/VS.Web.CG.Mvc/Identity/IdentityGeneratorFilesConfig.cs]: You're correct, adding OverWriteCondition.Never

File: src/VS.Web.CG.Mvc/Identity/identitygeneratorfilesconfig.json (hunk 2 above)
Comment (seancpeters) [reply src/VS.Web.CG.Mvc/Identity/identitygeneratorfilesconfig.json]: Site.js contains some comments, site.js.min is empty.

I think we should generate those files anyway, to have parity with templates.

Comment (javiercn) [reply src/VS.Web.CG.Mvc/Identity/identitygeneratorfilesconfig.json]: It's not a big deal, so I'm ok with it.

Comment (seancpeters) [reply src/VS.Web.CG.Mvc/Identity/identitygeneratorfilesconfig.json]: If you'd like me to remove it, I will. I've got a couple other changes to commit to this PR - it's not finalized yet.

File: src/VS.Web.CG.Mvc/VS.Web.CG.Mvc.csproj (hunk 1 above)
Comment (seancpeters) [reply src/VS.Web.CG.Mvc/VS.Web.CG.Mvc.csproj]: I'm also going to get rid of the `EmbeddedResourceRemove` and `None Remove` here, it's just VS cruft.

File: src/VS.Web.CG.Mvc/VS.Web.CG.Mvc.csproj [hunk 4]
```diff
@@ -42,9 +44,18 @@
...
+    <EmbeddedResource Remove="Templates\Identity\SupportPages\**" />
+    <None Remove="Templates\Identity\SupportPages\**" />
+  </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\VS.Web.CG\VS.Web.CG.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (seancpeters) [inline src/VS.Web.CG.Mvc/VS.Web.CG.Mvc.csproj]: I'll also get rid of this item group, more VS cruft.

**Comment by seancpeters** [comment]
The changes to the locations of support files and changes to the model caused some unit test failures. The 6th commit fixes the tests.

**Pull request merged**